File: SWEA/D2/1970. 쉬운 거스름돈/쉬운 거스름돈.py
coin = [50000, 10000, 5000, 1000, 500, 100, 50, 10]

T = int(input())

for x in range(1, T+1):
    result = []
    m = int(input())
    c = 0
    for i in coin:
        c = m // i
        m = m - (c * i)
        result.append(c)
    print("#%d" %(x))
    for j in result:
        print(j, end=" ")
    print()

File: SWEA/D2/1970. 쉬운 거스름돈/쉬운 거스름돈.java

import java.util.Scanner;
import java.io.FileInputStream;

public class Solution {
	
	public static void main(String args[]) throws Exception {
		
		Scanner sc = new Scanner(System.in);
		
		int test = sc.nextInt();
		
		for (int t = 1; t<=test; t++) {
			
			int num = sc.nextInt();
			
			int[] coin = {50000, 10000, 5000, 1000, 500, 100, 50, 10};
			int[] result = new int[8];
			
			for (int i = 0; i< 8 ; i++) {
				result[i] = num / coin[i];
				num %= coin[i];
			}
			
			System.out.println("#" + t);
			
			for (int i = 0; i < 8; i++) {
				System.out.print(result[i] + " ");
			}
			
			System.out.println();
		}
		
	}

}
